ACPs/196-precompile-secp256r1/README.md

```diff
@@ -0,0 +1,83 @@
+# ACP-196: Precompile for secp256r1 Curve Support
+
+| ACP | 196 |
+| :--- | :--- |
+| **Title** | Precompile for secp256r1 Curve Support |
+| **Author(s)** | [Santiago Cammi](https://github.com/scammi), [Arran Schlosberg](https://github.com/ARR4N)  |
+| **Status** | Proposed ([Discussion](https://github.com/avalanche-foundation/ACPs/discussions/XXX)) |
+| **Track** | Standards |
+
+## Abstract
+
+This proposal introduces a precompiled contract that performs signature verifications for the secp256r1 elliptic curve on Avalanche's C-Chain and Subnet-EVM chains. The precompile will be implemented at address `0x0000000000000000000000000000000000000100` and will enable native verification of P-256 signatures, significantly improving gas efficiency for biometric authentication systems, WebAuthn, and modern device-based signing mechanisms.
+
+## Motivation
+
+The secp256r1 (P-256) elliptic curve is the standard cryptographic curve used by modern device security systems, including Apple's Secure Enclave, Android Keystore, WebAuthn, and Passkeys. However, Avalanche currently only supports secp256k1 natively, forcing developers to use expensive Solidity-based verification that costs 200k-330k gas per signature verification.
+
+This ACP proposes implementing EIP-7212's secp256r1 precompiled contract to unlock significant ecosystem benefits:
+
+### **Enterprise & Institutional Adoption**
+- **Reduced onboarding friction**: Enterprises can leverage existing biometric authentication infrastructure instead of managing seed phrases or hardware wallets
+- **Regulatory compliance**: Institutions can utilize their approved device security standards and identity management systems
+- **Cost optimization**: 100x gas reduction (from 200k-330k to 3,450 gas) makes enterprise-scale applications economically viable
+
+The 100x gas cost reduction makes these use cases economically viable while maintaining the security properties institutions and users expect from their existing devices.
+
+## Specification
+
+This ACP implements [EIP-7212](https://eips.ethereum.org/EIPS/eip-7212) for secp256r1 signature verification on Avalanche. The specification follows EIP-7212 exactly, with the precompiled contract deployed at address `0x0000000000000000000000000000000000000100`.
+
+### Core Functionality
+- **Input**: 160 bytes (message hash + signature components r,s + public key coordinates x,y)
+- **Output**: 32 bytes (success: `0x...01`, failure: empty)  
```

Comment:
> **Output**: 32 bytes

This will be in the `returndata` buffer, which would typically be copied to memory and not the stack, so I don't think it's necessary to use an entire word as this increases the memory usage of the caller.

Reply:
Yes, I agree, but it breaks compatibility with other chains' EIP-7212 implementations.

Is this something we care about?

I understand that this should be a standalone document, so optimizing for efficiency makes sense.

Reply:
Good point. I'm in favour of ecosystem compatibility.

Do you know how many chains have implemented RIP-7212? With EIPs we always wait until they're live on Ethereum mainnet because then we know that it's a stable interface. I'd hate to release something, the rest of the ecosystem does something different, and we don't have shared tooling.

That said, we can always create a later ACP to add alternative interfaces if necessary. Same precompile, multiple ways to interact with it.